backend/cassandra/session.go

```diff
@@ -137,6 +137,34 @@ func (s *Session) GetTables(keyspace string) ([]string, error) {
 	return tables, nil
 }
 
+// GetTables queries the cassandra cluster for a list of an existing tables in a given keyspace.
+func (s *Session) GetVariables(ctx context.Context, query string) ([][]string, error) {
+	iter := s.session.Query(query).WithContext(ctx).Iter()
+
+    var variables [][]string
+
+	// check if one or two columns (are labels present?)
+	if (len(iter.Columns()) == 1) {
```

Comment:
I suggest to rewrite that if/else block in a more clean way. `else`s should be avoided as much as possible because they make code a bit dirty. Also, all validations should be done in the beginning. So, in this particular case:
```
if len(iter.Columns()) > 2 {
  return nil, fmt.Errorf("Expected 1 or 2 columns, got %d", len(iter.Columns()))
}

if (len(iter.Columns()) == 1) {
  ...
}

if (len(iter.Columns()) == 2) {
  ...
}
```

And it could be simplified even more if that logic
a) will be moved to `plugin` and
b) query will have different fields order(label, value -> value, label)
In that case these `if (len(iter.Columns()) ==` could be eradicated entirely:)

Reply:
Right, good idea